### ISSUE TITLE
Fail to read from s3 backend when using an assumed role inside a container (ecs)

### ISSUE THREAD

**Issue by daxroc** [hashicorp/terraform#14945]
Hi there,

I'm attempting to execute `terraform plan` within a container that has a task-role assigned. This role has sufficient privilege as I can read/write to the bucket in question.

### Terraform Version
version : 0.9.6

### Affected Resource(s)
s3 backend - possibly more as it fails fast here?

### Expected Behavior
What should have happened?
plan should succeed reading existing state from s3 backend

### Actual Behavior
plan failed due to unsuccessful request to s3 backend 

### Important Factoids
Running an ECS task with an assigned task role. exporting the credentials from `curl 169.254.170.2$AWS_CONTAINER_CREDENTIALS_RELATIVE_URI` as envvars `terraform plan` will succeed.

**Comment by daxroc** [comment]
I don't think this is an S3-backend bug rather a failure using the assumed credentials as it's not a typical usecase from inside a container.

**Comment by ingshtrom** [comment]
@daxroc how are you exporting the creds? I tried doing it and it didn't work for me. Am I missing something? 
![image](https://user-images.githubusercontent.com/571756/31277648-fc88a37e-aa6f-11e7-8141-dc40d146e475.png)

**Comment by daxroc** [comment]
Hey @ingshtrom I was doing something like -> https://gist.github.com/daxroc/4173792f79200db57296a24ce5600732

On a side note - packer seems not to need the above script

**Comment by greyfairer** [comment]
duplicate of #17056 ?

**Comment by markitoxs** [comment]
This is fixed on 0.12

**Comment by markitoxs** [comment]
It also works with 0.11.14
